src/Extensions/Hosting/Certificates/ICertificateReader.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.Omex.Extensions.Hosting.Certificates
+{
+	/// <summary>
+	/// Certificate reader interface
+	/// </summary>
+	public interface ICertificateReader
+	{
+		/// <summary>
+		/// Returns all certificates which match commonName parameter
+		/// </summary>
+		/// <param name="commonName">Common name of certificates</param>
+		/// <param name="refreshCache">Refresh cache before looking for certificate</param>
+		/// <param name="storeName">Certificate store</param>
+		IEnumerable<X509Certificate2> GetCertificatesByCommonName(string commonName, bool refreshCache = false, StoreName storeName = StoreName.My);
```

Comment:
>StoreName storeName = StoreName.My) [](start = 106, length = 35)

Would it be better to search in all stores if store is not specified? It will have a higher chance of finding the cert

Reply:
I don't think there is an option to specify something like `All` and just going thought all of them would probably be a bit overhead since we usually have them in a single store, but if user of a class needs it they could implement loop through all of it by themself